src/interfaces/controllers/get-trailer-registration.ts

```diff
@@ -0,0 +1,45 @@
+import { Request, Response, NextFunction } from 'express';
+import * as domain from '../../domain';
+import { getTrailerValidator } from '../validators/get-trailer-registration';
+import { ERRORS, GetTrailerRequest } from '../../domain';
+import { TrailerRegistrationBase } from './trailer-registration-base';
+import { debug } from '../../utils/logger';
+
+export class GetTrailerRegistration extends TrailerRegistrationBase {
+  private validate(payload: GetTrailerRequest): string {
+    const errors = getTrailerValidator.validate(payload).error;
+    return errors?.message;
+  }
+
+  public async call(req: Request, res: Response, next: NextFunction): Promise<void> {
+    const { vin } = req.params;
+    const { make } = req.query;
+    const payload: GetTrailerRequest = { vin, make: make as string };
+    const errors = this.validate(payload);
+    if (errors) {
+      next(new domain.HTTPError(400, errors));
+      return;
+    }
+
+    try {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+      const vinOrChassisWithMake = this.defineVinOrChassisWithMake(vin, payload.make);
+      const existingTrailerRegistration = await this.getTrailerRegistrationByVinOrChassisWithMake(vinOrChassisWithMake);
+      if (!existingTrailerRegistration) {
```

Comment:
we can probably simplify this further (and also don't believe the return is needed as it seems that we are passing the same thing to the middleware.
Will this work?
```js
if (!existingTrailerRegistration || existingTrailerRegistration.deregisterDate) {
 next(new domain.HTTPError(404, ERRORS.RECORD_NOT_FOUND))
}
```
Or I will need to have more clarity why we send 404 if we know we have deregistered something. Are we purposely hiding it for the client ? I would have thought that we would have a different message back to the user if it was the case.